recaf-core/src/main/java/me/coley/recaf/workspace/resource/source/DirectoryContentSource.java

```diff
@@ -58,4 +67,31 @@ protected String getPathName(Path entry) {
 		String absoluteEntry = entry.toAbsolutePath().toString();
 		return absoluteEntry.substring(absolutePath.length() + 1);
 	}
+	
+	private static final class PathByteSource implements ByteSource {
```

Comment:
Common cases like this can be extracted to the `io` package along with the actual `ByteSource` class.

The LocalFileHeader one isn't common so that I don't mind keeping as an inner class.